manual_benchmarks/test_datatypes_crunch.ipynb

```diff
@@ -0,0 +1,283 @@
+        "\n",
+        "from radis.io.hitemp import fetch_hitemp\n",
+        "from radis.db.classes import get_molecule\n",
+        "from radis.phys.constants import hc_k\n",
+        "\n",
+        "import numpy as np\n",
+        "from numpy import exp, pi"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "jhglpoCFXrpp"
+      },
+      "source": [
+        "### Function to reduce the memory usage of pandas dataframe"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "CryqdLGaXdFO"
+      },
+      "source": [
+        "def reduce_mem_usage(props):\n",
+        "    start_mem_usg = props.memory_usage().sum() / 1024**2 \n",
+        "    print(\"Memory usage of properties dataframe is :\",start_mem_usg,\" MB\")\n",
+        "    NAlist = [] # Keeps track of columns that have missing values filled in. \n",
+        "    for col in props.columns:\n",
+        "        if props[col].dtype != object:  # Exclude strings\n",
+        "            \n",
+        "            # Print current column type\n",
+        "            print(\"******************************\")\n",
+        "            print(\"Column: \",col)\n",
+        "            print(\"dtype before: \",props[col].dtype)\n",
+        "            \n",
+        "            # make variables for Int, max and min\n",
+        "            IsInt = False\n",
+        "            mx = props[col].max()\n",
+        "            mn = props[col].min()\n",
+        "            \n",
+        "            # Integer does not support NA, therefore, NA needs to be filled\n",
+        "            if not np.isfinite(props[col]).all(): \n",
+        "                NAlist.append(col)\n",
+        "                props[col].fillna(-1,inplace=True)  \n",
+        "                   \n",
+        "            # test if column can be converted to an integer\n",
+        "            asint = props[col].fillna(0).astype(np.int64)\n",
+        "            result = (props[col] - asint)\n",
+        "            result = result.sum()\n",
+        "            if result > -0.01 and result < 0.01:\n",
+        "                IsInt = True\n",
+        "\n",
+        "            \n",
+        "            # Make Integer/unsigned Integer datatypes\n",
+        "            if IsInt:\n",
+        "                if mn >= 0:\n",
+        "                    if mx < 255:\n",
+        "                        props[col] = props[col].astype(np.uint8)\n",
+        "                    elif mx < 65535:\n",
+        "                        props[col] = props[col].astype(np.uint16)\n",
+        "                    elif mx < 4294967295:\n",
+        "                        props[col] = props[col].astype(np.uint32)\n",
+        "                    else:\n",
+        "                        props[col] = props[col].astype(np.uint64)\n",
```

Comment:
For readability, I would write the limits as (1 << 8) - 1, (1 << 16) - 1, (1 << 32) -1, etc.
I believe that the -1 part is not even necessary, since it's integers and you're using "Lesser than" (without the "or equal to"), so you could use (1<<8), (1<<16), and (1<<32) as limits.

a << b means "bitshift integer a by b positions to the left", this means that 1 << b is shorthand for 2**b.